draft-protected-headers.md

```diff
@@ -28,58 +32,192 @@ author:
     email: dkg@fifthhorseman.net
 informative:
 normative:
+ RFC2119:
+ RFC2822:
  RFC3156:
+ RFC5751:
+ RFC8174:
+ RFC8551:
 --- abstract
 
-Cryptographic e-mails do not protect headers.  This document describes how several implementations protect headers.
+This document describes a common strategy to extend the cryptographic protections provided by PGP/MIME etc. to also protect message headers. In particular, how to encrypt the Subject line.
 
 --- middle
 
 Introduction
 ============
 
-FIXME
+The standards for cryptographic e-mail, PGP/MIME and S/MIME ({{RFC3156}} and {{RFC8551}}), can provide both integrity and confidentiality to the body of a MIME e-mail message. However, PGP/MIME alone provides no protection to message headers and the mechanism specified for header protection within S/MIME has not seen widespread adoption.
+
+This document defines a scheme, "Protected Headers for Cryptographic E-mail", which has been adopted by multiple existing e-mail clients in order to extend the cryptographic protections provided by PGP/MIME to also protect the {{RFC2822}} message headers.
+
+In particular, we describe how to encrypt the Subject line and how to preserve backwards compatibility so an encrypted subject remains available to recipients using software that does not implement support for the Protected Headers scheme.
+
+We also discuss some of the compatibility constraints and usability concerns which motivated the design of the scheme, as well as limitations.
+
+The authors believe the technique is broadly applicable would also apply to other MIME-compatible cryptographic e-mail systems, including S/MIME.  Furthermore, this technique has already proven itself as a useful building block for other improvements to cryptographic e-mail, such as the Autocrypt Level 1.1 "Gossip" mechanism.
+
 
 Requirements Language
 ---------------------
 
 The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in BCP 14 {{RFC2119}} {{RFC8174}} when, and only when, they appear in all capitals, as shown here.
 
-Cryptographic Payload
-=====================
 
-Message Composition (Header Copying)
-====================================
+Terminology
+-----------
+
+For the purposes of this document, we define the following concepts:
+
+   * *MUA* is short for Mail User Agent; an e-mail client.
+   * *Protection* of message data refers to cryptographic encryption and/or signatures, providing authenticity, confidentiality or both.
+   * *Cryptographic Envelope* is all MIME structure directly dictated by the cryptographic e-mail system in use.
```

Comment:
I'm trying to be abstract here, and I agree it makes things confusing. The cryptographic e-mail system could be OpenPGP/MIME, or S/MIME, or some other imaginary cryptosystem for MIME e-mail. See my other comment; I think maybe we should just stop trying to be abstract and make this concretely about OpenPGP/MIME, and add a future work section that proposes it might also be useful in other contexts.

I inherited the abstract direction from my interpretation of @dkg 's intent, so I'd like his feedback on whether we change course. It would simplify things though.

Reply:
I'm fine with the first draft focusing on PGP/MIME, and i think that will help us keep the document more narrowly targeted :)

It is important to me that we don't document it as though it could not be compatible with S/MIME though, and leave the door open to a future draft including S/MIME as well.